[zigbee] zigbee_basic_attrib_list / zigbee_identify_attrib_list not declared in generated main.cpp on nRF52/Zephyr

### The problem

When using the `zigbee` component on an nRF52840 board (Zephyr platform), compilation always fails with:

  error: 'zigbee_basic_attrib_list' was not declared in this scope
  error: 'zigbee_identify_attrib_list' was not declared in this scope

This affects every Zigbee entity type (binary_sensor, switch) — even a minimal config with a single entity fails. The generated `main.cpp` contains a cluster list (e.g. `zigbee_ep1_cluster_list[]`) that references `zigbee_basic_attrib_list` and `zigbee_identify_attrib_list`, but those declarations appear AFTER the cluster list in the generated file (or not at all), causing the compile error.

Looking at `zigbee_zephyr.py`, `_attr_to_code()` generates the attribute lists (via `CORE.add_global()`), and `zigbee_new_cluster_list()` also uses `CORE.add_global()` for the cluster list that references them. The ordering of these globals in the generated output places the cluster list before the attribute list declarations it depends on.


### Which version of ESPHome has the issue?

2026.2.1

### What type of installation are you using?

Home Assistant Add-on

### What platform are you using?

Other

### Component causing the issue

nRF52

### YAML Config

```yaml
esphome:
  name: zigbee-switch
  friendly_name: 4-Button Zigbee Switch

nrf52:
  board: adafruit_feather_nrf52840

zigbee:
  id: my_zigbee
  power_source: DC_SOURCE
  wipe_on_boot: once

logger:

output:
  - platform: gpio
    id: led_output
    pin: P0.31

light:
  - platform: binary
    id: status_led
    name: "Status LED"
    internal: true
    output: led_output

switch:
  - platform: template
    id: button_1_sw
    name: "Button 1"
    optimistic: true

binary_sensor:
  - platform: gpio
    id: btn1_gpio
    internal: true
    pin:
      number: P0.29
      mode: { input: true, pullup: true }
      inverted: true
    on_press:
      - switch.turn_on: button_1_sw
    on_release:
      - switch.turn_off: button_1_sw
```

### Anything in the logs that might be useful for us?

```txt
/data/build/zigbee-switch/src/main.cpp:83:115: error: 'zigbee_basic_attrib_list' was not declared in this scope; did you mean 'zigbee_ep1_attrib_list'?
   83 | zb_zcl_cluster_desc_t zigbee_ep1_cluster_list[] = {ZB_ZCL_CLUSTER_DESC(ZB_ZCL_CLUSTER_ID_BASIC, ZB_ZCL_ARRAY_SIZE(zigbee_basic_attrib_list, zb_zcl_attr_t), zigbee_basic_attrib_list, ...
/data/build/zigbee-switch/src/main.cpp:83:305: error: 'zigbee_identify_attrib_list' was not declared in this scope; did you mean 'zigbee_ep1_attrib_list'?
```

### Additional information

Tested with both `binary_sensor` and `switch` entity types — same error in both cases. The `adafruit_feather_nrf52840` board resolves correctly, Zephyr framework downloads and configures successfully, and compilation proceeds up to main.cpp before failing. The bug is purely in the ordering of CORE.add_global() calls in zigbee_zephyr.py — the attribute list declarations need to be emitted before the cluster list declarations that reference them.

--- a/esphome/components/zigbee/__init__.py
+++ b/esphome/components/zigbee/__init__.py
@@ -8,7 +8,7 @@ from esphome.components.zephyr import zephyr_add_pm_static, zephyr_data
 from esphome.components.zephyr.const import KEY_BOOTLOADER
 import esphome.config_validation as cv
 from esphome.const import CONF_ID, CONF_INTERNAL, CONF_NAME
-from esphome.core import CORE
+from esphome.core import CORE, CoroPriority, coroutine_with_priority
 from esphome.types import ConfigType
 
 from .const_zephyr import (
@@ -96,6 +96,7 @@ FINAL_VALIDATE_SCHEMA = cv.All(
 )
 
 
+@coroutine_with_priority(CoroPriority.CORE)
 async def to_code(config: ConfigType) -> None:
     cg.add_define("USE_ZIGBEE")
     if CORE.using_zephyr:

--- a/esphome/components/zigbee/zigbee_zephyr.py
+++ b/esphome/components/zigbee/zigbee_zephyr.py
@@ -179,6 +179,7 @@ async def zephyr_to_code(config: ConfigType) -> None:
                 "USE_ZIGBEE_WIPE_ON_BOOT_MAGIC", random.randint(0x000001, 0xFFFFFF)
             )
         cg.add_define("USE_ZIGBEE_WIPE_ON_BOOT")
+
     # Generate attribute lists before any await that could yield (e.g., build_automation
     # waiting for variables from other components). If the hub's priority decays while
     # yielding, deferred entity jobs may add cluster list globals that reference these

--- a/esphome/components/zigbee/zigbee_zephyr.py
+++ b/esphome/components/zigbee/zigbee_zephyr.py
@@ -179,6 +179,12 @@ async def zephyr_to_code(config: ConfigType) -> None:
                 "USE_ZIGBEE_WIPE_ON_BOOT_MAGIC", random.randint(0x000001, 0xFFFFFF)
             )
         cg.add_define("USE_ZIGBEE_WIPE_ON_BOOT")
+    # Generate attribute lists before any await that could yield (e.g., build_automation
+    # waiting for variables from other components). If the hub's priority decays while
+    # yielding, deferred entity jobs may add cluster list globals that reference these
+    # attribute lists before they're declared.
+    await _attr_to_code(config)
+
     var = cg.new_Pvariable(config[CONF_ID])
 
     if on_join_config := config.get(CONF_ON_JOIN):
@@ -186,7 +192,6 @@ async def zephyr_to_code(config: ConfigType) -> None:
 
     await cg.register_component(var, config)
 
-    await _attr_to_code(config)
     CORE.add_job(_ctx_to_code, config)
 
 

